### PR TITLE
vg/vggio: first import of a Gio-backed canvas

### PR DESCRIPTION
**Pull request by sbinet**
Updates gonum/plot#627.

### CODE REVIEW

**Comment by sbinet** [comment]
done.

**Comment by sbinet** [comment]
the lint failure (unable to inspect a dependency that uses `cgo`) is annoying.
it seems related to https://github.com/golangci/golangci-lint/issues/827 that has been closed...

not sure what to do at this point.

File: vg/vggio/vg.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+type Canvas struct {
+	*vgimg.Canvas
+	gtx layout.Context
+}
+
+const (
+	// DefaultDPI is the default dot resolution for image
+	// drawing in dots per inch.
+	DefaultDPI = 96
+)
```
Comment (kortschak) [inline vg/vggio/vg.go:33]:
```
// DefaultDPI is the default dot resolution for image
// drawing in dots per inch.
const DefaultDPI = 96
```

**Comment by kortschak** [comment]
The golangci-lint failure is irritating. I will file an issue to them about that later.

**Comment by kortschak** [comment]
I think I understand this failure; the GitHub action that runs golangci-lint does not have the Cgo dependencies installed (the travis build does, but it's independent). You will need to have pre-install step in a .golangci.yml file that installs those deps, see https://github.com/gonum/gonum/blob/master/.github/workflows/golangci-lint.yml and then adding something like

```
- name: cgo-dependencies
  run: sudo apt install pkg-config libgles2-mesa-dev libxkbcommon-dev libxkbcommon-x11-dev
```

**Comment by sbinet** [comment]
nice catch.

**Pull request merged**